src/qvm.lisp

```diff
@@ -177,9 +177,11 @@ This will not clear previously installed gates from the QVM."
   ;; Return the modified QVM.
   qvm)
 
-(defun compile-loaded-program (qvm)
+(defmethod compile-loaded-program ((qvm pure-state-qvm))
   "Compile the loaded program on the QVM QVM."
   (unless (program-compiled-p qvm)
+    (setf (program qvm) (quil::fuse-gates-in-executable-code (program qvm)))
```

Comment:
Am I reading this right that this feature is auto-on for compiled mode and auto-off otherwise?

Reply:
you're reading this right; it's only enabled in `-c`.